pennylane_lightning/core/src/gates/GateOperation.hpp

```diff
@@ -69,6 +69,32 @@ enum class GateOperation : uint32_t {
     END
 };
 
+enum class ControlledGateOperation : uint32_t {
+    BEGIN = 0,
+    /* Single-qubit gates */
+    PauliX = 0,
+    PauliY,
+    PauliZ,
+    Hadamard,
+    S,
+    T,
+    PhaseShift,
+    RX,
+    RY,
+    RZ,
+    /* Two-qubit gates */
+    SWAP,
+    IsingXX,
+    IsingXY,
+    IsingYY,
+    IsingZZ,
+    SingleExcitation,
+    SingleExcitationMinus,
+    SingleExcitationPlus,
+    /* END (placeholder) */
```

Comment:
do we want to add `DoubleExcitation` support?

Reply:
Might as well add it, but I have not done it yet because I have seen a single case of controlled excitation support in all demos and it was a controlled `SingleExcitation`. But sure, will do.

Reply:
I don't know if controlled DE is going to be a priority, but if it is an easy add, why not.